### ISSUE TITLE
ipv6 addresses in the ip() option of the syslog() source

### ISSUE THREAD

**Issue by czanik** [syslog-ng/syslog-ng#4701]
Someone reported, that the ```ip()``` option of the ```syslog()``` source does not work with IPv6 addresses:
```
czplaptop:/etc/syslog-ng/conf.d # syslog-ng -s
Error parsing within source, syntax error, unexpected ':', expecting LL_IDENTIFIER or LL_STRING in /etc/syslog-ng/conf.d/ipv6.conf:2:13-2:14:
1       source s_ipv6 {
2----->   syslog(ip(::1) ip-protocol(6) port(6514));
2----->             ^
3       };
4       destination d_ipv6 {
5         file("/var/log/ipv6");
6       };
7       log {source(s_ipv6); destination(d_ipv6);};

Included from /etc/syslog-ng/syslog-ng.conf:224:1-224:1:
```
It seems that IPv6 addresses only work if they are between quotes. IPv4 addresses do not require quotes.

I checked, the documentation does not mention that IPv6 addresses need to be between quotes. From the user's point of view it would be easier, if IPv6 addresses would also work without quotes, so I'd prefer to get it fixed in the code instead of the documentation.

**Comment by bazsi** [comment]
we are using ':' characters in other parts of the grammar and if we included it in unquoted strings that would not allow that.

For instance `@version: 4.4` the ':' would become part of the "version:" token. there are not many ':' characters today in the grammar, but I'd be reluctant to open up our LL_IDENTIFIER even more. Actually I've worked these past few days on an extended filter syntax where the number `#1` I needed to do was to remove the unquoted string behaviour completely as it prevents me to use a sane syntax.
